pressurecooker/html.py

```diff
@@ -0,0 +1,69 @@
+import os
+import tempfile
+import zipfile
+
+from bs4 import BeautifulSoup
+
+
+class HTMLParser:
+    def __init__(self, filename=None, html=None):
+        self.filename = filename
+        self.html = html
+        self.link_tags = {
+            'a': 'href',
+            'audio': 'src',
+            'img': 'src',
+            'link': 'href',
+            'script': 'src'
+        }
+
+    def get_links(self):
+        basename = None
+        if self.html is None:
+            basename = os.path.basename(self.filename)
+            self.html = open(self.filename).read()
+        soup = BeautifulSoup(self.html, 'html.parser')
+
+        extracted_links = []
+        for tag_name in self.link_tags:
+            tags = soup.find_all(tag_name)
+            for tag in tags:
+                link = tag.get(self.link_tags[tag_name])
+                # don't include links to ourselves or # links
+                # TODO: Should this part be moved to get_local_files instead?
+                if (basename and not link.startswith(basename)) and not link.strip().startswith("#"):
```

Comment:
One other case I've seen in a lot of chefs are urls that start with a `/` (e.g. `/css/stylesheet.css`)- does this handle this case?